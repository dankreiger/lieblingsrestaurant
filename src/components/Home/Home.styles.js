import styled from 'styled-components';
import { Container } from 'reactstrap';

export const HomeContainer = styled(Container)`
  padding-left: 0px !important;
  transition: padding 300ms;
  }
  &.toggled {
    padding-left: 250px !important;
    .arrow-icon {
      transform: rotate(180deg);
    }
  }
`;
