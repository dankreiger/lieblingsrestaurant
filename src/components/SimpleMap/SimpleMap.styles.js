import styled from 'styled-components';
import Geosuggest from 'react-geosuggest';
import { Popover } from 'reactstrap';
import {
  NAVIGATION_HEIGHT,
  DEFAULT_TRANSITION_DURATION
} from '../../constants';
import { DANGER_COLOR } from '../../constants/colorsConstants';

export const SearchInput = styled(Geosuggest)`
  &.geosuggest {
    font-size: 18px;
    font-size: 1rem;
    text-align: left;
    margin: 0 auto;
    width: 100vw;
    left: 0;
    position: absolute;
  }
  .geosuggest__suggests-wrapper {
    display: ${({ showSuggestions }) => (showSuggestions ? 'block' : 'none')};
  }
  .geosuggest__input {
    width: 100%;
    border: 2px solid
      ${({ invalidSelection }) =>
        invalidSelection ? DANGER_COLOR : 'transparent'};
    box-shadow: 0 0 1px #3d464d;
    padding: 0.5em 1em;
    transition: border 0.2s, box-shadow 0.2s;
  }
  .geosuggest__input:focus {
    border-color: #267dc0;
    box-shadow: 0 0 0 transparent;
  }
  .geosuggest__suggests {
    position: absolute;
    top: 100%;
    left: 0;
    right: 0;
    max-height: 25em;
    padding: 0;
    margin-top: -1px;
    background: #fff;
    border: 2px solid #267dc0;
    border-top-width: 0;
    overflow-x: hidden;
    overflow-y: auto;
    list-style: none;
    z-index: 5;
    -webkit-transition: max-height 0.2s, border 0.2s;
    transition: max-height 0.2s, border 0.2s;
  }
  .geosuggest__suggests--hidden {
    max-height: 0;
    overflow: hidden;
    border-width: 0;
  }

  /**
   * A geosuggest item
   */
  .geosuggest__item {
    font-size: 18px;
    font-size: 1rem;
    padding: 0.5em 0.65em;
    cursor: pointer;
  }
  .geosuggest__item:hover,
  .geosuggest__item:focus {
    background: #f5f5f5;
  }
  .geosuggest__item--active {
    background: #267dc0;
    color: #fff;
  }
  .geosuggest__item--active:hover,
  .geosuggest__item--active:focus {
    background: #ccc;
  }
  .geosuggest__item__matched-text {
    font-weight: bold;
  }
`;

export const StyledPopover = styled(Popover)`
  .popover-inner {
    border: 1px solid ${DANGER_COLOR};
  }
  .arrow {
    &:before {
      border-bottom-color: ${DANGER_COLOR} !important;
    }
  }
`;

export const MapContainer = styled.div`
  height: calc(100vh - ${NAVIGATION_HEIGHT}px);
  width: 100vw;
  position: fixed;
  top: ${NAVIGATION_HEIGHT}px;
  z-index: -1;
  left: 0;
  transition: padding ${DEFAULT_TRANSITION_DURATION}ms;
  padding-left: ${props => (props.toggled ? '250px' : '0px')};
`;
