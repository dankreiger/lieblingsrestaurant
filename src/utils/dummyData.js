export const dummyFavorites = [
  {
    description: 'Hotel Novotel Berlin Mitte, Fischerinsel, Berlin, Germany',
    isFixture: false,
    label: 'Hotel Novotel Berlin Mitte, Fischerinsel, Berlin, Germany',
    matchedSubstrings: { length: 1, offset: 6 },
    placeId: '1',
    gmaps: {
      address_components: [
        { long_name: '12', short_name: '12', types: ['street_number'] },
        {
          long_name: 'Fischerinsel',
          short_name: 'Fischerinsel',
          types: ['route']
        },
        {
          long_name: 'Mitte',
          short_name: 'Mitte',
          types: ['sublocality_level_1', 'sublocality', 'political']
        },
        {
          long_name: 'Berlin',
          short_name: 'Berlin',
          types: ['locality', 'political']
        },
        {
          long_name: 'Berlin',
          short_name: 'Berlin',
          types: ['administrative_area_level_1', 'political']
        },
        {
          long_name: 'Germany',
          short_name: 'DE',
          types: ['country', 'political']
        },
        { long_name: '10179', short_name: '10179', types: ['postal_code'] }
      ],
      adr_address:
        '<span class="street-address">Fischerinsel 12</span>, <span class="postal-code">10179</span> <span class="locality">Berlin</span>, <span class="country-name">Germany</span>',
      formatted_address: 'Fischerinsel 12, 10179 Berlin, Germany',
      formatted_phone_number: '030 206740',
      geometry: {
        location: { lat: 52.5130038, lng: 13.405086100000062 },
        viewport: {
          south: 52.51162996970849,
          west: 13.403695019708493,
          north: 52.51432793029149,
          east: 13.406392980291457
        }
      },
      icon: 'https://maps.gstatic.com/mapfiles/place_api/icons/lodging-71.png',
      id: '973e6e13b0f0cf6e845962f6bba43dee88384662',
      international_phone_number: '+49 30 206740',
      name: 'Hotel Novotel Berlin Mitte',
      opening_hours: {
        open_now: true,
        periods: [
          {
            open: {
              day: 0,
              time: '0000',
              hours: 0,
              minutes: 0,
              nextDate: 1541890800000
            }
          }
        ],
        weekday_text: [
          'Monday: Open 24 hours',
          'Tuesday: Open 24 hours',
          'Wednesday: Open 24 hours',
          'Thursday: Open 24 hours',
          'Friday: Open 24 hours',
          'Saturday: Open 24 hours',
          'Sunday: Open 24 hours'
        ]
      },
      photos: [
        {
          height: 3205,
          html_attributions: [
            '<a href="https://maps.google.com/maps/contrib/111482740722870294072/photos">Hotel Novotel Berlin Mitte</a>'
          ],
          width: 2103
        },
        {
          height: 2414,
          html_attributions: [
            '<a href="https://maps.google.com/maps/contrib/111482740722870294072/photos">Hotel Novotel Berlin Mitte</a>'
          ],
          width: 3620
        },
        {
          height: 3464,
          html_attributions: [
            '<a href="https://maps.google.com/maps/contrib/117126757311301328454/photos">Stefan Thomas</a>'
          ],
          width: 4618
        },
        {
          height: 2414,
          html_attributions: [
            '<a href="https://maps.google.com/maps/contrib/111482740722870294072/photos">Hotel Novotel Berlin Mitte</a>'
          ],
          width: 3620
        },
        {
          height: 2103,
          html_attributions: [
            '<a href="https://maps.google.com/maps/contrib/111482740722870294072/photos">Hotel Novotel Berlin Mitte</a>'
          ],
          width: 3219
        },
        {
          height: 1879,
          html_attributions: [
            '<a href="https://maps.google.com/maps/contrib/111482740722870294072/photos">Hotel Novotel Berlin Mitte</a>'
          ],
          width: 4650
        },
        {
          height: 768,
          html_attributions: [
            '<a href="https://maps.google.com/maps/contrib/111482740722870294072/photos">Hotel Novotel Berlin Mitte</a>'
          ],
          width: 1024
        },
        {
          height: 2414,
          html_attributions: [
            '<a href="https://maps.google.com/maps/contrib/111482740722870294072/photos">Hotel Novotel Berlin Mitte</a>'
          ],
          width: 3620
        },
        {
          height: 2413,
          html_attributions: [
            '<a href="https://maps.google.com/maps/contrib/111482740722870294072/photos">Hotel Novotel Berlin Mitte</a>'
          ],
          width: 3620
        },
        {
          height: 2413,
          html_attributions: [
            '<a href="https://maps.google.com/maps/contrib/111482740722870294072/photos">Hotel Novotel Berlin Mitte</a>'
          ],
          width: 3620
        }
      ],
      place_id: '1',
      plus_code: {
        compound_code: 'GC74+62 Berlin, Germany',
        global_code: '9F4MGC74+62'
      },
      rating: 4.3,
      reference: '1',
      reviews: [
        {
          author_name: 'Richard Williams',
          author_url:
            'https://www.google.com/maps/contrib/104457413771304589923/reviews',
          language: 'en',
          profile_photo_url:
            'https://lh6.googleusercontent.com/-FSRrczyiA-A/AAAAAAAAAAI/AAAAAAAAAE4/LrPB2lEfNq8/s128-c0x00000000-cc-rp-mo-ba3/photo.jpg',
          rating: 4,
          relative_time_description: 'a week ago',
          text:
            'Hotel staff are friendly and polite. Hotel is very clean and modern. There is a small area for children to hang out. Food menu was a little limited but what was on offer was good. Drink prices are typical of a city hotel. Bed was comfy enough. Overall very good hotel.',
          time: 1540457363
        },
        {
          author_name: 'Toxic Flame',
          author_url:
            'https://www.google.com/maps/contrib/115742897245149551845/reviews',
          language: 'en',
          profile_photo_url: 'https://google.com/photo.jpg',
          rating: 4,
          relative_time_description: 'in the last week',
          text:
            'Everything was excellent as I wished. The hotelroom was nice, quiet and peaceful. They had a nice bathroom with everything I could think of. The breakfast was really good and local too. But the room was a little small for 3 people.',
          time: 1541070364
        },
        {
          author_name: 'Daisy Motta Santos',
          author_url:
            'https://www.google.com/maps/contrib/113208788533793089337/reviews',
          language: 'en',
          profile_photo_url:
            'https://lh6.googleusercontent.com/-wZj6XfOYyhE/AAAAAAAAAAI/AAAAAAAAfjU/Fjkv6B4cq54/s128-c0x00000000-cc-rp-mo-ba4/photo.jpg',
          rating: 5,
          relative_time_description: 'a month ago',
          text:
            'Great hotel with a perfect restaurant! The hotel is really nice and the localization is very useful. The restaurant Novo2 is perfect. Delicious food. The only thing that I need deserve to improve is the gym facility. Only one small treadmill. The parking cost 19 euro per day and you can go out and come back without pay any extra. The bathroom amenities are perfect! Great shower and reasonable bed!',
          time: 1537786888
        },
        {
          author_name: "Chris O'Mara",
          author_url:
            'https://www.google.com/maps/contrib/110266619940291502573/reviews',
          language: 'en',
          profile_photo_url: 'https://google.com/photo.jpg',
          rating: 5,
          relative_time_description: '3 weeks ago',
          text:
            'Great service, lovely room, impressed with the eco-friendly touches. Generous with assistance too. Will definitely stay again',
          time: 1539624914
        },
        {
          author_name: 'James Wang',
          author_url: 'https://www.google.com/reviews',
          language: 'en',
          profile_photo_url: 'https://google.com/photo.jpg',
          rating: 4,
          relative_time_description: '2 weeks ago',
          text:
            'Nice and quiet hotel. It served my needs for a comfortable and quiet stay.',
          time: 1539843668
        }
      ],
      scope: 'GOOGLE',
      types: [
        'lodging',
        'restaurant',
        'point_of_interest',
        'food',
        'establishment'
      ],
      url: 'https://maps.google.com/',
      utc_offset: 60,
      vicinity: 'Fischerinsel 12, Berlin',
      website: 'http://www.hotel.com',
      html_attributions: []
    },
    location: { lat: 52.5130038, lng: 13.405086100000062 },
    photo: 'https://placebear.com/g/965/787',
    photos: [
      'https://placebear.com/g/965/787',
      'https://placebear.com/g/965/787',
      'https://placebear.com/g/965/787',
      'https://placebear.com/g/965/787',
      'https://placebear.com/g/965/787',
      'https://placebear.com/g/965/787',
      'https://placebear.com/g/965/787',
      'https://placebear.com/g/965/787',
      'https://placebear.com/g/965/787',
      'https://placebear.com/g/965/787',
      'https://placebear.com/g/965/787',
      'https://placebear.com/g/965/787'
    ],
    favorited: true,
    rating: 1
  },
  {
    description: 'Hotel Novotel Berlin Mitte, Fischerinsel, Berlin, Germany',
    isFixture: false,
    label: 'Hotel Novotel Berlin Mitte, Fischerinsel, Berlin, Germany',
    matchedSubstrings: { length: 1, offset: 6 },
    placeId: '2',
    gmaps: {
      address_components: [
        { long_name: '12', short_name: '12', types: ['street_number'] },
        {
          long_name: 'Fischerinsel',
          short_name: 'Fischerinsel',
          types: ['route']
        },
        {
          long_name: 'Mitte',
          short_name: 'Mitte',
          types: ['sublocality_level_1', 'sublocality', 'political']
        },
        {
          long_name: 'Berlin',
          short_name: 'Berlin',
          types: ['locality', 'political']
        },
        {
          long_name: 'Berlin',
          short_name: 'Berlin',
          types: ['administrative_area_level_1', 'political']
        },
        {
          long_name: 'Germany',
          short_name: 'DE',
          types: ['country', 'political']
        },
        { long_name: '10179', short_name: '10179', types: ['postal_code'] }
      ],
      adr_address:
        '<span class="street-address">Fischerinsel 12</span>, <span class="postal-code">10179</span> <span class="locality">Berlin</span>, <span class="country-name">Germany</span>',
      formatted_address: 'Fischerinsel 12, 10179 Berlin, Germany',
      formatted_phone_number: '030 206740',
      geometry: {
        location: { lat: 52.5130038, lng: 13.405086100000062 },
        viewport: {
          south: 52.51162996970849,
          west: 13.403695019708493,
          north: 52.51432793029149,
          east: 13.406392980291457
        }
      },
      icon: 'https://maps.gstatic.com/mapfiles/place_api/icons/lodging-71.png',
      id: '973e6e13b0f0cf6e845962f6bba43dee88384662',
      international_phone_number: '+49 30 206740',
      name: 'Hotel Novotel Berlin Mitte',
      opening_hours: {
        open_now: true,
        periods: [
          {
            open: {
              day: 0,
              time: '0000',
              hours: 0,
              minutes: 0,
              nextDate: 1541890800000
            }
          }
        ],
        weekday_text: [
          'Monday: Open 24 hours',
          'Tuesday: Open 24 hours',
          'Wednesday: Open 24 hours',
          'Thursday: Open 24 hours',
          'Friday: Open 24 hours',
          'Saturday: Open 24 hours',
          'Sunday: Open 24 hours'
        ]
      },
      photos: [
        {
          height: 3205,
          html_attributions: [
            '<a href="https://maps.google.com/maps/contrib/111482740722870294072/photos">Hotel Novotel Berlin Mitte</a>'
          ],
          width: 2103
        },
        {
          height: 2414,
          html_attributions: [
            '<a href="https://maps.google.com/maps/contrib/111482740722870294072/photos">Hotel Novotel Berlin Mitte</a>'
          ],
          width: 3620
        },
        {
          height: 3464,
          html_attributions: [
            '<a href="https://maps.google.com/maps/contrib/117126757311301328454/photos">Stefan Thomas</a>'
          ],
          width: 4618
        },
        {
          height: 2414,
          html_attributions: [
            '<a href="https://maps.google.com/maps/contrib/111482740722870294072/photos">Hotel Novotel Berlin Mitte</a>'
          ],
          width: 3620
        },
        {
          height: 2103,
          html_attributions: [
            '<a href="https://maps.google.com/maps/contrib/111482740722870294072/photos">Hotel Novotel Berlin Mitte</a>'
          ],
          width: 3219
        },
        {
          height: 1879,
          html_attributions: [
            '<a href="https://maps.google.com/maps/contrib/111482740722870294072/photos">Hotel Novotel Berlin Mitte</a>'
          ],
          width: 4650
        },
        {
          height: 768,
          html_attributions: [
            '<a href="https://maps.google.com/maps/contrib/111482740722870294072/photos">Hotel Novotel Berlin Mitte</a>'
          ],
          width: 1024
        },
        {
          height: 2414,
          html_attributions: [
            '<a href="https://maps.google.com/maps/contrib/111482740722870294072/photos">Hotel Novotel Berlin Mitte</a>'
          ],
          width: 3620
        },
        {
          height: 2413,
          html_attributions: [
            '<a href="https://maps.google.com/maps/contrib/111482740722870294072/photos">Hotel Novotel Berlin Mitte</a>'
          ],
          width: 3620
        },
        {
          height: 2413,
          html_attributions: [
            '<a href="https://maps.google.com/maps/contrib/111482740722870294072/photos">Hotel Novotel Berlin Mitte</a>'
          ],
          width: 3620
        }
      ],
      place_id: '2',
      plus_code: {
        compound_code: 'GC74+62 Berlin, Germany',
        global_code: '9F4MGC74+62'
      },
      rating: 4.3,
      reference: '2',
      reviews: [
        {
          author_name: 'Richard Williams',
          author_url:
            'https://www.google.com/maps/contrib/104457413771304589923/reviews',
          language: 'en',
          profile_photo_url:
            'https://lh6.googleusercontent.com/-FSRrczyiA-A/AAAAAAAAAAI/AAAAAAAAAE4/LrPB2lEfNq8/s128-c0x00000000-cc-rp-mo-ba3/photo.jpg',
          rating: 4,
          relative_time_description: 'a week ago',
          text:
            'Hotel staff are friendly and polite. Hotel is very clean and modern. There is a small area for children to hang out. Food menu was a little limited but what was on offer was good. Drink prices are typical of a city hotel. Bed was comfy enough. Overall very good hotel.',
          time: 1540457363
        },
        {
          author_name: 'Toxic Flame',
          author_url:
            'https://www.google.com/maps/contrib/115742897245149551845/reviews',
          language: 'en',
          profile_photo_url: 'https://google.com/photo.jpg',
          rating: 4,
          relative_time_description: 'in the last week',
          text:
            'Everything was excellent as I wished. The hotelroom was nice, quiet and peaceful. They had a nice bathroom with everything I could think of. The breakfast was really good and local too. But the room was a little small for 3 people.',
          time: 1541070364
        },
        {
          author_name: 'Daisy Motta Santos',
          author_url:
            'https://www.google.com/maps/contrib/113208788533793089337/reviews',
          language: 'en',
          profile_photo_url:
            'https://lh6.googleusercontent.com/-wZj6XfOYyhE/AAAAAAAAAAI/AAAAAAAAfjU/Fjkv6B4cq54/s128-c0x00000000-cc-rp-mo-ba4/photo.jpg',
          rating: 5,
          relative_time_description: 'a month ago',
          text:
            'Great hotel with a perfect restaurant! The hotel is really nice and the localization is very useful. The restaurant Novo2 is perfect. Delicious food. The only thing that I need deserve to improve is the gym facility. Only one small treadmill. The parking cost 19 euro per day and you can go out and come back without pay any extra. The bathroom amenities are perfect! Great shower and reasonable bed!',
          time: 1537786888
        },
        {
          author_name: "Chris O'Mara",
          author_url:
            'https://www.google.com/maps/contrib/110266619940291502573/reviews',
          language: 'en',
          profile_photo_url: 'https://google.com/photo.jpg',
          rating: 5,
          relative_time_description: '3 weeks ago',
          text:
            'Great service, lovely room, impressed with the eco-friendly touches. Generous with assistance too. Will definitely stay again',
          time: 1539624914
        },
        {
          author_name: 'James Wang',
          author_url: 'https://www.google.com/reviews',
          language: 'en',
          profile_photo_url: 'https://google.com/photo.jpg',
          rating: 4,
          relative_time_description: '2 weeks ago',
          text:
            'Nice and quiet hotel. It served my needs for a comfortable and quiet stay.',
          time: 1539843668
        }
      ],
      scope: 'GOOGLE',
      types: [
        'lodging',
        'restaurant',
        'point_of_interest',
        'food',
        'establishment'
      ],
      url: 'https://maps.google.com/',
      utc_offset: 60,
      vicinity: 'Fischerinsel 12, Berlin',
      website: 'http://www.hotel.com',
      html_attributions: []
    },
    location: { lat: 52.5130038, lng: 13.405086100000062 },
    photo: 'https://placebear.com/g/965/787',
    photos: [
      'https://placebear.com/g/965/787',
      'https://placebear.com/g/965/787',
      'https://placebear.com/g/965/787',
      'https://placebear.com/g/965/787',
      'https://placebear.com/g/965/787',
      'https://placebear.com/g/965/787',
      'https://placebear.com/g/965/787',
      'https://placebear.com/g/965/787',
      'https://placebear.com/g/965/787',
      'https://placebear.com/g/965/787',
      'https://placebear.com/g/965/787',
      'https://placebear.com/g/965/787'
    ],
    favorited: true,
    rating: 1
  }
];
