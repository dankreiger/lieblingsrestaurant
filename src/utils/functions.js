export const pluralize = (word, items) => {
  if (!items) {
    return `0 ${word}s`;
  } else if (items.length === 1) {
    return `1 ${word}`;
  } else {
    return `${items.length} ${word}s`;
  }
};

export const getPhoto = item => {
  if (item.photo) {
    return item.photo;
  } else if (item.gmaps) {
    return item.gmaps.photos ? item.gmaps.photos[0].getUrl() : item.gmaps.icon;
  } else {
    return null;
  }
};

export const fetchAllPhotos = item => {
  if (item.gmaps && item.gmaps.photos) {
    return item.gmaps.photos.map(photo => photo.getUrl());
  } else if (item.gmaps.icon) {
    return item.gmaps.icon;
  } else {
    return null;
  }
};
