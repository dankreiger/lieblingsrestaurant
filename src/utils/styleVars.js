import { NAVIGATION_HEIGHT } from '../constants';

export const fixedTopButton = () =>
  `
  background: #f57c00;
  font-size: 2em;
  display: flex;
  align-items: center;
  height: ${NAVIGATION_HEIGHT}px;
  justify-content: center;
  cursor: pointer;
  color: #fff;
  box-shadow: 1px 1px 3px rgba(0, 0, 0, 0.4);
  position: relative;
  overflow: hidden;
  flex-grow: 1;
  &:after {
    content: '';
    position: absolute;
    top: 50%;
    left: 50%;
    width: 5px;
    height: 5px;
    background: rgba(255, 255, 255, 0.5);
    opacity: 0;
    border-radius: 100%;
    transform: scale(1, 1) translate(-50%);
    transform-origin: 50% 50%;
  }

  &:focus {
    outline: none;
  }

  &:focus:not(:active)::after {
    animation: ripple 1s ease-out;
  }
`;
