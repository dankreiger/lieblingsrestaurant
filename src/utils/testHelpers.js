import React from 'react';
import { Provider } from 'react-redux';
import { mount, render, shallow } from 'enzyme';
import { BrowserRouter as Router } from 'react-router-dom';
import configureStore from 'redux-mock-store';

const enzymeRenderComponent = (component, store, enzymeFunction) => {
  switch (enzymeFunction) {
    case 'mount':
      return mount(<Provider store={store}>{component}</Provider>);
    case 'shallow':
      return shallow(<Provider store={store}>{component}</Provider>);
    default:
      return render(<Provider store={store}>{component}</Provider>);
  }
};

const enzymeRenderComponentWithRouter = (component, store, enzymeFunction) => {
  switch (enzymeFunction) {
    case 'mount':
      return mount(
        <Provider store={store}>
          <Router>{component}</Router>
        </Provider>
      );
    case 'shallow':
      return shallow(
        <Provider store={store}>
          <Router>{component}</Router>
        </Provider>
      );
    default:
      return render(
        <Provider store={store}>
          <Router>{component}</Router>
        </Provider>
      );
  }
};

export const setupReduxConnectedComponent = (
  component,
  enzymeFunction,
  favorites = [],
  navigation = { toggled: false }
) => {
  const middlewares = [];
  const mockStore = configureStore(middlewares);
  const store = mockStore({ favorites, navigation });

  return enzymeRenderComponent(component, store, enzymeFunction);
};

export const setupReduxConnectedComponentWithRouter = (
  component,
  enzymeFunction,
  favorites = [],
  navigation = { toggled: false }
) => {
  const middlewares = [];
  const mockStore = configureStore(middlewares);
  const store = mockStore({ favorites, navigation });

  return enzymeRenderComponentWithRouter(component, store, enzymeFunction);
};
