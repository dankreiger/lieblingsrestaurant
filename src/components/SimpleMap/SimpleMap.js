/*global google */

import React, { Component } from 'react';
import { connect } from 'react-redux';
import GoogleMapReact from 'google-map-react';
import { apiIsLoaded, appendGmapScript } from './helpers/gmapFunctions';

import MapMarker from '../MapMarker/MapMarker';
import { SearchInput } from './SimpleMap.styles';
import { BERLIN } from '../../constants';
import { getPhoto } from '../../utils/functions';

class SimpleMap extends Component {
  state = {
    places: [],
    currentMapInfo: null,
    mapReady: false
  };

  componentDidMount() {
    if (!window.google) {
      let s = appendGmapScript();

      s.addEventListener('load', () => {
        this.setState({ mapReady: true });
      });
    } else {
      this.setState({ mapReady: true });
    }
    this.setState({ places: this.props.favorites });
  }

  handleMapInstance = (map, maps, places) => {
    apiIsLoaded(map, maps, places);
    this.setState({ currentMapInfo: { map, maps, places } });
  };

  handleSuggestSelect = newPlace => {
    if (!newPlace) {
      return;
    }
    const { map, maps, places } = this.state.currentMapInfo;

    if (!places.some(place => place.placeId === newPlace.placeId)) {
      this.setState({ places: [...places, newPlace] }, () => {
        this.handleMapInstance(map, maps, this.state.places);
      });
    }
  };

  render() {
    const { places, currentMapInfo, mapReady } = this.state;
    return (
      <>
        {currentMapInfo && (
          <SearchInput
            placeholder="Search for a restaurant..."
            onSuggestSelect={this.handleSuggestSelect}
            location={new google.maps.LatLng(BERLIN.lat, BERLIN.lng)}
            radius={2000}
            types={['establishment']}
          />
        )}
        <div style={{ height: '70vh', width: '100%' }}>
          {mapReady && (
            <GoogleMapReact
              defaultCenter={{ lat: BERLIN.lat, lng: BERLIN.lng }}
              defaultZoom={12}
              yesIWantToUseGoogleMapApiInternals
              options={{ disableDoubleClickZoom: true }}
              onGoogleApiLoaded={({ map, maps }) =>
                this.handleMapInstance(map, maps, places)
              }
            >
              {places.map(place => (
                <MapMarker
                  key={place.placeId}
                  text={place.label}
                  icon={place.icon || place.gmaps.icon}
                  photo={getPhoto(place)}
                  lat={place.location.lat}
                  lng={place.location.lng}
                  favorite={place}
                  favorited={place.favorited}
                />
              ))}
            </GoogleMapReact>
          )}
        </div>
      </>
    );
  }
}

const mapStateToProps = ({ favorites }) => ({ favorites });

export default connect(
  mapStateToProps,
  null
)(SimpleMap);
