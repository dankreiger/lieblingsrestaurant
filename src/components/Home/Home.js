import React from 'react';
import { func } from 'prop-types';
import { connect } from 'react-redux';
import * as actions from './../../actions';
import { Row, Col } from 'reactstrap';
import SimpleMap from '../SimpleMap/SimpleMap';
import { HomeContainer } from './Home.styles';
import classNames from 'classnames';
import NavigationToggler from '../NavigationToggler/NavigationToggler';
import { favoritesTypes } from '../../reducers/types/favorites.types';
import { navigationTypes } from '../../reducers/types/navigation.types';

const Home = ({ favorites, toggleNavigation, navigation }) => (
  <HomeContainer fluid className={classNames({ toggled: navigation.toggled })}>
    <Row noGutters>
      <Col className="d-flex">
        {favorites.length > 0 && (
          <NavigationToggler
            favorites={favorites}
            toggleNavigation={toggleNavigation}
          />
        )}
      </Col>
      <Col
        xs={favorites.length > 0 ? 8 : 12}
        sm={favorites.length > 0 ? 10 : 12}
      >
        <SimpleMap />
      </Col>
    </Row>
  </HomeContainer>
);

Home.propTypes = {
  ...favoritesTypes,
  ...navigationTypes,
  toggleNavigation: func
};

const mapStateToProps = ({ favorites, navigation }) => ({
  favorites,
  navigation
});

export default connect(
  mapStateToProps,
  actions
)(Home);
