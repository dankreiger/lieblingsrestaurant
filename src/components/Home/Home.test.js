import React from 'react';
import { shallow } from 'enzyme';

import Home from './Home';
import Root from '../../Root';
import SimpleMap from '../SimpleMap/SimpleMap';
import { setupGoogleMock } from '../../utils/mocks';

describe('Home', () => {
  let homeComponent;

  beforeAll(() => {
    setupGoogleMock();
  });

  beforeEach(() => {
    homeComponent = shallow(<Home />);
  });
  describe('rendering', () => {
    test('renders as expected', () => {
      expect(homeComponent).toBeTruthy();
      expect(homeComponent.length).toBe(1);
      expect(homeComponent).toMatchSnapshot();
    });
  });

  describe('structure', () => {
    test('renders 1 map component', () => {
      expect(homeComponent.find(SimpleMap)).toBeTruthy();
      expect(homeComponent.find(SimpleMap).length).toBe(1);
    });
  });
});
