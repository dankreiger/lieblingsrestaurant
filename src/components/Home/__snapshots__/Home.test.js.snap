// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home rendering renders as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <Row
        tag="div"
      >
        <Col
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Connect(SimpleMap) />
        </Col>
      </Row>,
      "fluid": true,
      "tag": "div",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Col
          tag="div"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ]
          }
        >
          <Connect(SimpleMap) />
        </Col>,
        "tag": "div",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Connect(SimpleMap) />,
          "tag": "div",
          "widths": Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Row
          tag="div"
        >
          <Col
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <Connect(SimpleMap) />
          </Col>
        </Row>,
        "fluid": true,
        "tag": "div",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Col
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <Connect(SimpleMap) />
          </Col>,
          "tag": "div",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Connect(SimpleMap) />,
            "tag": "div",
            "widths": Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
